source/common/ssl/utility.cc

```diff
@@ -1,8 +1,12 @@
 #include "common/ssl/utility.h"
```

Comment:
Note: This file doesn't exist anymore. @bdecoste moved TLS socket to extensions, so that it can be compiled out (and another TLS library can be used in it's place), it's new home is in `source/extensions/transport_sockets/tls`.

Having said that, I'd prefer if you created another target for this (e.g. `source/extensions/common/crypto`?), since those functions have nothing to do with SSL/TLS.

Reply:
I was actually thinking about making a "crypto" util but just ran with the suggestion to put it in common/ssl/utillty. Easy change :)

Reply:
I think this should be in core actually, since even core code might want to compute HMAC and SHAs. How about `source/common/crypto`?

Reply:
Right now I don't leverage any code that is different between Open/BoringSSL, but it would require some compile-time plumbing, which is not as clean as swapping in/out an extension, if neither lib is used or some other crypto impl was used instead.

How about I move this to `source/common/crypto` for now. And make an issue to propose adding a crypto impl extension api?

Reply:
Sure, that sounds good. Would like to merge this PR soon, so whatever is reasonable here and we can then look at the next one, thanks.